### PR TITLE
Upgrade Beam version to 2.55.1

### PR DESCRIPTION
**Pull request by Abacn**

### CODE REVIEW

**Comment by Abacn** [comment]
Python and xlang template upgrade deferred in this PR due to https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/1331#discussion_r1559549723

**Comment by Abacn** [comment]
Many Yaml template and Python template tests failing: 

```
YAMLTemplateIT.testSimpleCompositeSpec: YAMLTemplateIT#L55
The job failed before launch! For more information please check the job log at https://console.cloud.google.com/dataflow/jobs/us-central1/2024-04-10_08_10_53-11488117926144804100?project=cloud-teleport-testing.

YAMLTemplateIT.testSimpleCompositeSpecFile: YAMLTemplateIT#L64
The job failed before launch! For more information please check the job log at https://console.cloud.google.com/dataflow/jobs/us-central1/2024-04-10_08_10_53-6297951897493783767?project=cloud-teleport-testing.

PubSubToBigQueryIT.testPubsubToBigQueryWithPythonUdf: PubSubToBigQueryIT#L122
The job failed before launch! For more information please check the job log at https://console.cloud.google.com/dataflow/jobs/us-central1/2024-04-10_09_40_12-15803571501679049692?project=cloud-teleport-testing.
TextIOToBigQueryIT.testTextIOToBigQueryWithPythonUdf: TextIOToBigQueryIT#L100
The job failed before launch! For more information please check the job log at https://console.cloud.google.com/dataflow/jobs/us-central1/2024-04-10_09_41_57-7236990968357205595?project=cloud-teleport-testing.
TextToBigQueryStreamingIT.testTextToBigQueryWithPythonUDFs: TextToBigQueryStreamingIT#L102
The job failed before launch! For more information please check the job log at https://console.cloud.google.com/dataflow/jobs/us-central1/2024-04-10_09_41_19-12324055143971730472?project=cloud-teleport-testing.
PubsubProtoToBigQueryIT.pubSubProtoToBigQueryWithPythonUdf: PubsubProtoToBigQueryIT#L172
The job failed before launch! For more information please check the job log at https://console.cloud.google.com/dataflow/jobs/us-central1/2024-04-10_08_54_36-8171764657624851257?project=cloud-teleport-testing.
```

**Comment by Abacn** [comment]
```
2024-04-10 11:58:00.752 EDT
  File "/root/.apache_beam/cache/venvs/py-3.11-beam-2.55.1-da39a3ee5e6b4b0d3255bfef95601890afd80709/lib/python3.11/site-packages/apache_beam/internal/pickler.py", line 31, in <module>
2024-04-10 11:58:00.752 EDT
    from apache_beam.internal import cloudpickle_pickler
2024-04-10 11:58:00.752 EDT
  File "/root/.apache_beam/cache/venvs/py-3.11-beam-2.55.1-da39a3ee5e6b4b0d3255bfef95601890afd80709/lib/python3.11/site-packages/apache_beam/internal/cloudpickle_pickler.py", line 36, in <module>
2024-04-10 11:58:00.753 EDT
    import cloudpickle
2024-04-10 11:58:00.753 EDT
ModuleNotFoundError: No module named 'cloudpickle'
```

File: v1/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerConfig.java [hunk 1]
```diff
@@ -86,6 +87,8 @@ public abstract class SpannerConfig implements Serializable {
 
   public abstract @Nullable ValueProvider<Boolean> getDataBoostEnabled();
 
+  public abstract @Nullable ValueProvider<Credentials> getCredentials();
```
Comment (damccorm) [inline v1/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerConfig.java:90]: Why do we need these spanner changes?

**Review by damccorm — APPROVED** [review]

**Comment by Abacn** [comment]
Merged master onto PR to retrigger tests. Previously "CLA/Google" was stuck

The Dataflow Integration Test failing YamlTemplateIT is expected (also happens on main)

**Pull request merged**